hw11-weak-hashmap/src/main/java/ru/otus/hibernate/dao/ObjectDaoHibernate.java

```diff
@@ -0,0 +1,126 @@
+package ru.otus.hibernate.dao;
+
+
+import org.hibernate.Session;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.otus.cachehw.HwCache;
+import ru.otus.core.dao.ObjectDao;
+import ru.otus.core.dao.ObjectDaoException;
+import ru.otus.core.sessionmanager.SessionManager;
+import ru.otus.hibernate.sessionmanager.DatabaseSessionHibernate;
+import ru.otus.hibernate.sessionmanager.SessionManagerHibernate;
+
+import javax.persistence.Id;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ObjectDaoHibernate<T> implements ObjectDao<T> {
+    private static final Logger logger = LoggerFactory.getLogger(ObjectDaoHibernate.class);
+
+    private final Class<T> clazz;
+    private final List<Field> idFields;
+    private final List<Field> nIdFields;
+
+    private final HwCache<Optional,T> cache;
```

Comment:
Использовать Optional в качестве ключа как-то странно.
Лучше сразу конкретный тип:
```suggestion
    private final HwCache<Integer,T> cache;
```

Или параметоризовать ObjectDaoHibernate / ObjectDao
```
ObjectDaoHibernate<K, T>
ObjectDao<K, T>
```